### PR TITLE
Fix the ingestion error of snowflake on Pod label fields

### PR DESCRIPTION
**Pull request by dreamtalen**
In this commit, we fix the ingestion error of snowpipe on flow records with Pod label fields. The reason behind this error is commas are used to separate different columns in CSV and Pod label columns contain commas so they are separated into multiple columns when ingested into snowflake.
To fix it, we add a customized file format for flow records and use single quote enclosing Pod label fields.

Changes on Antrea side: https://github.com/antrea-io/antrea/pull/4334

### CODE REVIEW

**Review by heanlan — APPROVED** [review]
LGTM

File: snowflake/pkg/infra/stack.go
Comment (antoninbas) [inline snowflake/pkg/infra/stack.go:295]: I don't think you need the `pulumi.DependsOn` property here so I suggest removing it altogether to avoid confusion.
The only dependency is on the `schema` and that's already ensured by the `Schema` arg.

**Review by antoninbas — COMMENTED** [review]
You think that defining a custom format is neater than just using `FIELD_OPTIONALLY_ENCLOSED_BY` in the `COPY` statement? I tend to agree as it is easier to manage if we need extra customizations in the future.

File: snowflake/pkg/infra/stack.go
Comment (dreamtalen) [reply snowflake/pkg/infra/stack.go:295]: I didn't add any dependencies at first, but encountered an error when creating the pipe said `File format 'FLOWFORMAT' does not exist or not authorized.`, althrough previous log has shown `FileFormat antrea-sf-flows-format created`. After testing serveral rounds, I found adding these two dependencies could solve this issue.
I agree this looks confusing, do you have any suggestions I could try?

Comment (antoninbas) [reply snowflake/pkg/infra/stack.go:295]: It's a bit of a shot in the dark but first, I'd recommend using the full file format name ("<db>.<schema>.<name>") in the copy statement. I think that by adding these dependencies, you may be affecting session parameters and hiding a bug. Just like I am using the full name for the stage I think you need the full stage for the file format.

Comment (dreamtalen) [reply snowflake/pkg/infra/stack.go]: Thanks for advices! I just tested out the simpler way (adding that option in copy into statement directly). Think twice I agree it would be better to have a custom file format if we have multiple customizations to do in the future. For now it may be a little over engineering.

Comment (antoninbas) [reply snowflake/pkg/infra/stack.go]: > I agree it would be better to have a custom file format if we have multiple customizations to do in the future. For now it may be a little over engineering.

The thing is we have to keep the upgrade process robust. Pulumi is supposed to help with that, but the Snowflake provider is not always perfect. Switching from inline options to separate file format may be a bit tricky. I'll let you decide as I think additional customizations are not that likely. I still think you should give a try to the full name solution, it may just fix the issue.

Comment (dreamtalen) [reply snowflake/pkg/infra/stack.go]: Yes the full name solution solves that issue. Sorry I only edited my last comment and didn't send a new one.

**Review by antoninbas — APPROVED** [review]
LGTM, assuming you no longer want to introduce a separate file format resource

**Pull request merged**
